test/jdk/java/net/Authenticator/B4769350.java

```diff
@@ -355,7 +355,7 @@ static void proxyReply (HttpExchange exchange, String reply)
                 throws IOException
         {
             exchange.getResponseHeaders().add("Proxy-Authenticate", reply);
-            exchange.sendResponseHeaders(407, 0);
+            exchange.sendResponseHeaders(407, -1);
```

Comment:
Similarly here:

```java
exchange.sendResponseHeaders(407, -1 /* no response body */);
```

Reply:
I think a better solution would be to add a constant to the api class - rather than adding comments everywhere. E.g NO_CONTENT=-1 and UNLIMITED_CONTENT=0 or INDETERMINENT_CONTENT=0

Reply:
> I think a better solution would be to add a constant to the api class - rather than adding comments everywhere. E.g NO_CONTENT=-1 and UNLIMITED_CONTENT=0 or INDETERMINENT_CONTENT=0, or CHUNKED_CONTENT=0

That's not a bad idea. I am doing a small update to the API under another bug id and I might include that in it. I kind of agree that adding comments might be overkill. We could probably update a lot of the tests mechanically to use these symbolic names afterwards.

Reply:
I’m on mobile now so I haven’t looked at the api - but I think adding default methods like

sendResponseHeadersNoContent(int code)

and OutputStream sendRedsponseHeadersChunked(int code) 

would also make the api easier to use.

Reply:
I am more than willing to work on an API change PR to add these methods. It is an abstract class, so they could collide with existing code, but I am guessing it is a trivial fix if it does, and it would break at compile time not runtime.

adding protected constants to the HttpExchange abstract class seems trivially safe.

Reply:
I created a PR for the api changes https://github.com/openjdk/jdk/pull/18955

Reply:
> I created a PR for the api changes #18955

I am working on a PR with some other API changes. I'd prefer to include this change in that one as well. API changes need to be approved through the CSR process as well as code review.

Reply:
Understood. I only wanted to get down on paper what I proposed.